pkg/scheduler/scheduler_test.go

```diff
@@ -492,17 +493,16 @@ func TestSchedulerErrorWithLongBinding(t *testing.T) {
 			client := clientsetfake.NewSimpleClientset(&node)
 			informerFactory := informers.NewSharedInformerFactory(client, 0)
 			predicateMap := map[string]predicates.FitPredicate{"PodFitsHostPorts": predicates.PodFitsHostPorts}
-
 			scheduler, bindingChan := setupTestSchedulerLongBindingWithRetry(
 				queuedPodStore, scache, informerFactory, predicateMap, stop, test.BindingDuration)
-
+			scheduler.mu.Lock()
 			informerFactory.Start(stop)
 			informerFactory.WaitForCacheSync(stop)
-
 			scheduler.Run()
+
 			queuedPodStore.Add(firstPod)
 			queuedPodStore.Add(conflictPod)
-
+			scheduler.mu.Unlock()
```

Comment:
IMO, we're waiting for `queuePodStore` here, we just need to make sure `queuePodStore` is thread-safe instead of `Run()`.